validator/client/src/main/java/tech/pegasys/teku/validator/client/ValidatorClientService.java

```
@@ -219,6 +213,57 @@ public static ValidatorClientService create(
+      final AsyncRunner asyncRunner,
+      final boolean generateEarlyAttestations,
+      final boolean preferSszBlockEncoding,
+      final boolean failoversSendSubnetSubscriptions,
+      final Duration beaconNodeEventStreamSyncingStatusQueryPeriod) {
+    final BeaconNodeApi beaconNodeApi;
+    if (validatorConfig.getSentryNodeConfigurationFile().isEmpty()) {
+      beaconNodeApi =
+          validatorConfig
+              .getBeaconNodeApiEndpoints()
+              .map(
+                  beaconNodeApiEndpoints ->
+                      RemoteBeaconNodeApi.create(
+                          services,
+                          asyncRunner,
+                          beaconNodeApiEndpoints,
+                          config.getSpec(),
+                          generateEarlyAttestations,
+                          preferSszBlockEncoding,
+                          failoversSendSubnetSubscriptions,
+                          beaconNodeEventStreamSyncingStatusQueryPeriod))
+              .orElseGet(
+                  () ->
+                      InProcessBeaconNodeApi.create(
+                          services, asyncRunner, generateEarlyAttestations, config.getSpec()));
+    } else {
+      beaconNodeApi =
+          validatorConfig
+              .getSentryNodeConfigurationFile()
+              .map((sentryConfigPath) -> new SentryNodesConfigLoader().load(sentryConfigPath))
+              .map(
+                  (sentryNodesConfig) ->
+                      SentryBeaconNodeApi.create(
+                          services,
+                          asyncRunner,
+                          sentryNodesConfig,
+                          config.getSpec(),
+                          generateEarlyAttestations,
+                          preferSszBlockEncoding,
+                          failoversSendSubnetSubscriptions,
+                          beaconNodeEventStreamSyncingStatusQueryPeriod))
+              .orElseThrow(RuntimeException::new);
```

We should probably throw an exception with a more meaningful error message here.  Though it may be better to just write this starting with a `validatorConfig.getSentryNodeConfigurationFile().get()` and skip the map calls given we've already done the isEmpty() check above?